src/cpu_vehicles_camera_path.c

```diff
@@ -0,0 +1,4565 @@
+                temp_a2 = -temp_a2;
+            }
+            if (arg1_times_8 < 24) {
+                interp = gLapCompletionPercentByPlayerId[playerId];
+                D_80163150[playerId] = temp_a3[8] * interp + temp_a3[0] * (1.0f - interp);
+            } else {
+                D_80163150[playerId] = temp_a3[0];
+            }
+            if (D_80163150[playerId] < temp_a2 && player->speed / 18.0f * 216.0f >= 20.0f) {
+                return false;
+            }
+            return true;
+        }
+        progress = gNumPathPointsTraversed[playerId] - gNumPathPointsTraversed[gBestRankedHumanPlayer];
+        rank = gGPCurrentRaceRankByPlayerId[2 + (gBestRankedHumanPlayer * 4)];
+        if (gPathCountByPathIndex[0] * 2 / 3 < progress && rank >= 6) {
+            progress = gNumPathPointsTraversed[playerId] - gNumPathPointsTraversed[gLapCountByPlayerId[-26 + rank * 2]];
+        }
+        if (progress < 0) {
+            progress = -progress;
+        }
+        if (arg1_times_8 < 24) {
+            interp = gLapCompletionPercentByPlayerId[playerId];
+            var_a0_2 = temp_a3[8] * interp + temp_a3[0] * (1.0f - interp);
+        } else {
+            var_a0_2 = temp_a3[0];
+        }
+        D_80163150[playerId] = (gCCSelection + 1) * var_a0_2;
+        if (D_80163150[playerId] < progress && player->speed / 18.0f * 216.0f >= 20.0f) {
+            return false;
+        }
+        return true;
+    } else {
+        var_a1_4 = gNumPathPointsTraversed[(s16) *D_80163344] - gNumPathPointsTraversed[playerId];
+        if (var_a1_4 < 0) {
+            var_a1_4 = -var_a1_4;
+        }
+        var_v0 = 0;
+        for (i = 0; i < 2; i++) {
+            if (gGPCurrentRaceRankByPlayerId[D_80163344[i]] < arg2) {
+                var_v0++;
+            }
+        }
+        var_a0_4 = 0;
+        for (i = 0; i < gPlayerCount; i++) {
+            if (gGPCurrentRaceRankByPlayerId[i] < arg2) {
+                var_a0_4++;
+            }
+        }
+        var_t1 = (arg2 - var_v0) - var_a0_4;
+        if (var_v0 > 0 || var_a0_4 > 0) {
+            var_t1++;
+        }
+        D_80164538[playerId] = var_t1;
+        if (var_t1 < 0 || var_t1 >= 8) {
+            return false;
+        }
+        if (arg1_times_8 < 24) {
+            temp_a3 = &temp_a3[var_t1];
+            interp = gLapCompletionPercentByPlayerId[playerId];
+            D_80163150[playerId] = temp_a3[8] * interp + temp_a3[0] * (1.0f - interp);
+        } else {
+            D_80163150[playerId] = temp_a3[var_t1];
+        }
+        D_80163128[playerId] = var_a1_4;
+        if (D_80163150[playerId] < var_a1_4) {
+            return true;
+        }
+        return false;
+    }
+}
+#else
+GLOBAL_ASM("asm/non_matchings/cpu_vehicles_camera_path/func_800088D8.s")
+#endif
+
+void set_current_path(s32 pathIndex) {
+    gCurrentTrackPath = gTrackPaths[pathIndex];
+    gCurrentTrackInnerPath = gTrackInnerPath[pathIndex];
+    gCurrentTrackOuterPath = gTrackOuterPath[pathIndex];
```

Comment:
Are you sure about this inner outter path? Are they actually separated paths?

Reply:
yes